src/login/logind-core.c

```diff
@@ -526,6 +526,24 @@ int manager_count_displays(Manager *m) {
                 if (!streq_ptr(udev_device_get_subsystem(p), "drm"))
                         continue;
 
+                nn = udev_device_get_sysname(d);
+                if (!nn)
+                        continue;
+
+                /* Ignore internal displays: the type is encoded in
+                 * the sysfs name, as the second das seperated item
+                 * (the first is the card name, the last the connector
+                 * number). Internal displays are LVDS and eDP, hence
+                 * ignore specifically these types. */
+                dash = strchr(nn, '-');
+                if (dash && (startswith(dash, "-LVDS-") || startswith(dash, "-eDP-")))
+                        continue;
```

Comment:
I know, I was the one who told you to use a blacklist for external connectors, but I'm not sure this is the way to go. Below, I pasted the list of known connector types (excluding 'subconnectors'). I'm not at all sure about `9PinDIN` and `DSI` (but I think both are external, exclusively). Controversial, though, is how to treat "virtual". Generally, I'd say we should treat anything as "internal" which we didn't explicitly detect as external. In other words, always suspend except if we explicitly know an external display is connected. This basically makes we wonder whether we should instead use a whitelist for external connectors (again, I know, I said something different before.. sorry!).

Comments? I really think the default (and fallback!) should be to suspend, except if we're 100% clear we know what we're doing.

```
#define DRM_MODE_CONNECTOR_Unknown	0
#define DRM_MODE_CONNECTOR_VGA		1
#define DRM_MODE_CONNECTOR_DVII		2
#define DRM_MODE_CONNECTOR_DVID		3
#define DRM_MODE_CONNECTOR_DVIA		4
#define DRM_MODE_CONNECTOR_Composite	5
#define DRM_MODE_CONNECTOR_SVIDEO	6
#define DRM_MODE_CONNECTOR_LVDS		7
#define DRM_MODE_CONNECTOR_Component	8
#define DRM_MODE_CONNECTOR_9PinDIN	9
#define DRM_MODE_CONNECTOR_DisplayPort	10
#define DRM_MODE_CONNECTOR_HDMIA	11
#define DRM_MODE_CONNECTOR_HDMIB	12
#define DRM_MODE_CONNECTOR_TV		13
#define DRM_MODE_CONNECTOR_eDP		14
#define DRM_MODE_CONNECTOR_VIRTUAL      15
#define DRM_MODE_CONNECTOR_DSI		16
```